### PR TITLE
New module Az.DataBox

### PR DESCRIPTION
**Pull request by dhawal777**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/993}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by VeryEarly** [comment]
hi @dhawal777 ,

Please checkout your branch from:
https://github.com/Azure/azure-powershell/tree/generation

and make your PR against this branch.
thanks

**Comment by dhawal777** [comment]
> hi @dhawal777 ,
> 
> Please checkout your branch from:
> https://github.com/Azure/azure-powershell/tree/generation
> 
> and make your PR against this branch.
> thanks

Sure @VeryEarly, Just can you please tell why PR from checkout branch preferred over fork branch

**Comment by VeryEarly** [comment]
> > hi @dhawal777 ,
> > Please checkout your branch from:
> > https://github.com/Azure/azure-powershell/tree/generation
> > and make your PR against this branch.
> > thanks
> 
> Sure @VeryEarly, Just can you please tell why PR from checkout branch preferred over fork branch

For generated modules, all the development are on generation branch which contains readme.md/docs/tests/custom only, you only check in these necessary files into this branch. After PR was merged to generation branch, we will generate everything from ADO to migrate to main branch.

**Comment by dhawal777** [comment]
> > > hi @dhawal777 ,
> > > Please checkout your branch from:
> > > https://github.com/Azure/azure-powershell/tree/generation
> > > and make your PR against this branch.
> > > thanks
> > 
> > 
> > Sure @VeryEarly, Just can you please tell why PR from checkout branch preferred over fork branch
> 
> For generated modules, all the development are on generation branch which contains readme.md/docs/tests/custom only, you only check in these necessary files into this branch. After PR was merged to generation branch, we will generate everything from ADO to migrate to main branch.

Yep @VeryEarly idea is clear but sorry ask is still not clear, I have raised PR on generation branch only(myRepo:generation => AzurePowershell:generation), with only required folders , so it will get merged into generation branch only, what is wrong, please highlight

**Comment by VeryEarly** [comment]
Hi @dhawal777 ,

Sorry I thought your branch was forked from main branch, my bad.
Will review now.

**Comment by dhawal777** [comment]
> Hi @dhawal777 ,
> 
> Sorry I thought your branch was forked from main branch, my bad.
> 
> For the build error, please check in the Az.Databox.psd1 with this PR (should be generated in your local)

CheckedIn

**Comment by VeryEarly** [comment]
please check the error log:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=1032581&view=logs&jobId=e1b7b984-9f58-529f-7c5a-f15f8e35cfa6&j=e1b7b984-9f58-529f-7c5a-f15f8e35cfa6&t=e8689ae8-b216-55f7-b207-4831f84ff98a

you can rename "ProxyUseDefaultCredentials" -> "ProxyUseDefaultCredential" for get/remove/stop/update AzDataBoxJob/AzDataBoxJobCredential.

And suppress other warning messages.

**Comment by dhawal777** [comment]
> please check the error log:
> https://dev.azure.com/azure-sdk/public/_build/results?buildId=1032581&view=logs&jobId=e1b7b984-9f58-529f-7c5a-f15f8e35cfa6&j=e1b7b984-9f58-529f-7c5a-f15f8e35cfa6&t=e8689ae8-b216-55f7-b207-4831f84ff98a
> 
> you can rename "ProxyUseDefaultCredentials" -> "ProxyUseDefaultCredential" for get/remove/stop/update AzDataBoxJob/AzDataBoxJobCredential.
> 
> And suppress other warning messages.

@VeryEarly , Is this not right way to rename this parameters

- where:
      verb: New
      parameter-name: ProxyUseDefaultCredentials
    set:
      parameter-name: ProxyUseDefaultCredential

**Comment by VeryEarly** [comment]
> > please check the error log:
> > https://dev.azure.com/azure-sdk/public/_build/results?buildId=1032581&view=logs&jobId=e1b7b984-9f58-529f-7c5a-f15f8e35cfa6&j=e1b7b984-9f58-529f-7c5a-f15f8e35cfa6&t=e8689ae8-b216-55f7-b207-4831f84ff98a
> > you can rename "ProxyUseDefaultCredentials" -> "ProxyUseDefaultCredential" for get/remove/stop/update AzDataBoxJob/AzDataBoxJobCredential.
> > And suppress other warning messages.
> 
> @VeryEarly , Is this not right way to rename this parameter for one cmdlet
> 
> * where:
>   verb: New
>   parameter-name: ProxyUseDefaultCredentials
>   set:
>   parameter-name: ProxyUseDefaultCredential

looks correct to me. I think you can try remove the verb in this case since all "ProxyUseDefaultCredentials" will be renamed

**Comment by dhawal777** [comment]
Thanks @VeryEarly for suggestion but not sure why even for individual cmdlet it is not working

`[Parameter(DontShow)]
    [Microsoft.Azure.PowerShell.Cmdlets.DataBox.Category('Runtime')]
    [System.Management.Automation.SwitchParameter]
    # Use the default credentials for the proxy
    ${ProxyUseDefaultCredentials}`

**Comment by dhawal777** [comment]
Hey @VeryEarly, Please call out what I am missing in getting right generation for this property. Explored options in directive.md but didn't got any success.

**Comment by VeryEarly** [comment]
> Hey @VeryEarly, Please call out what I am missing in getting right generation for this property. Explored options in directive.md but didn't got any success.

Hi @dhawal777 ,

We discovered it's a bug in our pipeline and was fixed now, please sync your branch with `generation`, no action need to be taken for parameter `ProxyUseDefaultCredentials`

**Comment by dhawal777** [comment]
> > Hey @VeryEarly, Please call out what I am missing in getting right generation for this property. Explored options in directive.md but didn't got any success.
> 
> Hi @dhawal777 ,
> 
> We discovered it's a bug in our pipeline and was fixed now, please sync your branch with `generation`, no action need to be taken for parameter `ProxyUseDefaultCredentials`

Thanks @VeryEarly for the update.

**Comment by dhawal777** [comment]
Hi @VeryEarly , Now it is failing with need shouldprocess based on verb(New) cmdlet is using but "model cmdlet" which just form a readable way of creating a hashmap, may not need that, any suggestion here?

**Comment by VeryEarly** [comment]
> Hi @VeryEarly , Now it is failing with need shouldprocess based on verb(New) cmdlet is using but "model cmdlet" which just form a readable way of creating a hashmap, may not need that, any suggestion here?

You're right, not need to add support shouldprocess, just suppress these messages in 
https://github.com/dhawal777/azure-powershell/blob/generation/tools/StaticAnalysis/Exceptions/Az.DataBox/SignatureIssues.csv

File: src/DataBox/custom/Az.DataBox.custom.psm1
Comment (VeryEarly) [inline src/DataBox/custom/Az.DataBox.custom.psm1:1]: no need to check in this file, please remove

**Comment by VeryEarly** [comment]
Hi @dhawal777 , PR looks good except the psm1 file I commented, please remove and I'll merge.

**Comment by dhawal777** [comment]
> Hi @dhawal777 , PR looks good except the psm1 file I commented, please remove and I'll merge.

Thanks @VeryEarly for review and support, Will be re-reviewing example description once and push if any change required, Kindly keep it open till then, will comment for merge, Also what is release cycle PowerShell follow, like when we can leverage this changes, also if any update require in future how much time it take for it to go live, Also if they any other document for does and don't for next version update, kindly share it as well

**Comment by VeryEarly** [comment]
Please find our release plan here: https://github.com/Azure/azure-powershell/milestones

You can remove the "Do Not Merge" label once PR was ready and let me know.
Thanks.

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**
